resources/puppetlabs/lein-ezbake/template/foss/ext/debian/postinst.erb

```diff
@@ -4,12 +4,16 @@
 
 # On upgrade, we should restart the service if it's running
 if [ $1 = 'configure' -a -n $2 ] ; then
-  if [ -x '/etc/init.d/<%= EZBake::Config[:project] %>' ] ; then
-    # Using sysv
-    invoke-rc.d <%= EZBake::Config[:project] %> try-restart || :
-  elif [ -x /bin/systemctl ] ; then
+  if [ -e "/lib/systemd/system/<%= EZBake::Config[:project] -%>.service" ] ; then
     # Using systemd
+    if [ -e '/etc/init.d/<%= EZBake::Config[:project] %>' ] ; then
+      # We need to clean up the old sysv files if they still exist
+      rm '/etc/init.d/<%= EZBake::Config[:project] %>'
+    fi
     systemctl daemon-reload >/dev/null 2>&1 || :
-    systemctl try-restart <%= EZBake::Config[:project] %>.service
+    systemctl try-restart <%= EZBake::Config[:project] %>.service ||:
```

Comment:
It looks like puppet-agent is always a runtime dependency of ezbake packages. I'm going to follow @shrug 's advice and record the state of the service we are upgrading using puppet, and then apply that specific state to ensure consistency